chapters/categorical-predictors.qmd

```diff
@@ -0,0 +1,718 @@
+
+Note that linear dependencies are exact linear relationships between predictors. These are other circumstances where two or more predictors are highly correlated but not perfectly related. This situation, called **multicolinearity**, is discussed throughout the book but mostly in @sec-colinearity. 
+
+In addition to reference cell and one-hot encodings, there are various tools for creating indicator columns. Some, called _contrast methods_, are associated with more niche strategies, and many produce fractional values instead of binary indicators. For example, Helmert contrasts [@ruberg] are configured so that each category is compared to the average of the preceding categories. No encoding method is uniformly better than the others; the choice is often driven by how the user will interpret the parameter estimates. 
+
+The situation becomes slightly more complex when a model contains multiple categorical predictors. The reference value combines levels across predictors for reference value parameterizations. For example, if a model included the customer's country of residence, the reference would include that factor. For example, if Canadians were the reference nationality, the overall reference cell would be Canadian contract customers. 
+
+Note that these encoding techniques are basically look-up tables that map individual categories known in the training set to specific numeric values. As discussed in the next section, they can be problematic when novel values are seen in new data (such as the test set). 
+
+## Novel Categories {#sec-novel-categories}
+
+When creating indicator variables, there is the assumption that the categories in the training data reflect all possible categories. This may not be the case. For example, the agents observed in our data are not the only agents that will be booking rooms in the future. After we create a model with a specific set of agents, what happens when we predict future data that involves new agents? 
+
+One approach we could take is to prospectively create a new value for the predictor (let's give it a value of `"new"`). When we have new agents, we could assign them to this general category so that the model can anticipate this new level. In general, this won't help since the model estimation process will never have any "new" values on hand when the model is fit. For some models, this is innocuous and has no real harm (but offers no benefit either). 
```

Comment:
```suggestion
One approach we could take is to prospectively create a new value for the predictor (let's give it a value of `"new"`). When a future sample has an agent that was not in the original data, we could assign this sample to the general category so that the model can anticipate this new level. In general, this won't help since the model estimation process will never have any "new" values on hand when the model is fit. For some models, this is innocuous and has no real harm (but offers no benefit either). 
```